x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/elasticsearch/ElasticsearchInternalServiceSettings.java

```diff
@@ -42,36 +82,112 @@ public static ElasticsearchInternalServiceSettings fromMap(Map<String, Object> m
             ADAPTIVE_ALLOCATIONS,
             validationException
         );
-        String modelId = extractRequiredString(map, MODEL_ID, ModelConfigurations.SERVICE_SETTINGS, validationException);
+
+        // model id is optional as the ELSER and E5 service will default it
+        String modelId = extractOptionalString(map, MODEL_ID, ModelConfigurations.SERVICE_SETTINGS, validationException);
+
+        if (numAllocations == null && adaptiveAllocationsSettings == null) {
+            validationException.addValidationError(
+                ServiceUtils.missingOneOfSettingsErrorMsg(
+                    List.of(NUM_ALLOCATIONS, ADAPTIVE_ALLOCATIONS),
+                    ModelConfigurations.SERVICE_SETTINGS
+                )
+            );
+        }
 
         // if an error occurred while parsing, we'll set these to an invalid value, so we don't accidentally get a
         // null pointer when doing unboxing
-        return new ElasticsearchInternalServiceSettings(
-            numAllocations,
-            Objects.requireNonNullElse(numThreads, FAILED_INT_PARSE_VALUE),
-            modelId,
-            adaptiveAllocationsSettings
-        );
+        return new ElasticsearchInternalServiceSettings.Builder().setNumAllocations(numAllocations)
+            .setNumThreads(Objects.requireNonNullElse(numThreads, FAILED_INT_PARSE_VALUE))
+            .setModelId(modelId)
+            .setAdaptiveAllocationsSettings(adaptiveAllocationsSettings);
     }
 
     public ElasticsearchInternalServiceSettings(
         Integer numAllocations,
         int numThreads,
-        String modelVariant,
+        String modelId,
         AdaptiveAllocationsSettings adaptiveAllocationsSettings
     ) {
-        super(numAllocations, numThreads, modelVariant, adaptiveAllocationsSettings);
+        this.numAllocations = numAllocations;
+        this.numThreads = numThreads;
+        this.modelId = modelId;
```

Comment:
How about we wrap this in `Objects.requireNonNull()` to ensure that it's never constructed with null. If null was passed we'd have problems when serializing.